sdcm/cluster_k8s/__init__.py

```diff
@@ -395,9 +403,9 @@ def deploy_cert_manager(self, pool_name: str = None) -> None:
             LOGGER.debug(self.helm(
                 f"install cert-manager jetstack/cert-manager --version v{self.params.get('k8s_cert_manager_version')}",
                 namespace="cert-manager", values=helm_values))
-            time.sleep(10)
 
-        self.kubectl("wait --timeout=10m --all --for=condition=Ready pod", namespace="cert-manager")
+        self.kubectl_wait("--all --for=condition=Ready pod", namespace="cert-manager",
+                          timeout=600)
```

Comment:
Honestly, I don't see any reason to replace one method call with another where both do the same. Both run "kubectl wait" which ends up earlier than timeout if everything is ok.
"Sleep" above is needed to reach the situation where pods "appear", removing that sleep you can reach the situation where pods were not created yet and nothing will be checked at all in such case.
I agree that 10seconds may be too huge value. 2 or 3 sec would be enough.

Reply:
reason is that if there no such pods you are running kubectl wait and there are no pods that are matched it will fail, as result whole run fails